src/point.rs

```diff
@@ -0,0 +1,129 @@
+use std::borrow::Borrow;
+
+#[derive(Hash, Debug, Clone, Copy, PartialEq, Eq)]
+pub struct Point {
+    pub x: i32,
+    pub y: i32,
+}
+
+impl Point {
+    pub fn new(x: i32, y: i32) -> Point {
+        Point {x, y}
+    }
+
+    pub fn manhaten_distance(&self) -> u32 {
+        self.x.abs() as u32 + self.y.abs() as u32
+    }
+}
+
+
+pub fn get_closest<I, T>(points: I) -> Option<T>
+        where I: Iterator<Item = T>,
+              T: Borrow<Point> {
```

Comment:
Nice 😄 

you can do it with one type parameter less, though 😛 
```suggestion
pub fn closest<I: IntoIterator>(points: I) -> Option<<I as IntoIterator>::Item>
        where <I as IntoIterator>::Item: Borrow<Point> {
```

Reply:
since we're only using `points` in the `for` loop, we only need it to be `IntoIterator` (this worked so far because `Iterator` automatically implements `IntoIterator` by returning itself)

Reply:
I actually think that it is more readable and understandable with 2 generic types

Reply:
Hmm. It appears the type casting is unnecessary. what do you think of this:
```rust
pub fn closes<I: IntoIterator>(points: I) -> Option<I::Item>
    where I::Item: Borrow<T> { ... }
```